### PR TITLE
VAULT-26819 - implement hvs apps read

### PR DESCRIPTION
**Pull request by mercedesbh**
### Changes proposed in this PR:
This PR adds the `read` command for hvs apps for ticket https://hashicorp.atlassian.net/browse/VAULT-26819

### How I've tested this PR:
Ran `make go/build` to manually test local changes, and added unit tests.

### How I expect reviewers to test this PR:
1. Checkout this branch
2. Run make go/build
3. Create an app `./bin/hcp vault-secrets apps read {YOUR_APP_NAME}`

Screenshot 2024-05-16 at 9 27 47 AM

### Checklist:
- [x] Tests added if applicable
- [ ] CHANGELOG entry added or label 'pr/no-changelog' added to PR
  > Run `CHANGELOG_PR=<PR number> make changelog/new-entry` for guidance
  > in authoring a changelog entry, and commit the resulting file, which should
  > have a name matching your PR number. Entries should use imperative present
  > tense (e.g. Add support for...)

### CODE REVIEW

**Review by dadgar — APPROVED** [review]

**Review by briankassouf — APPROVED** [review]

File: internal/commands/vaultsecrets/apps/read.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
+package apps
```
Comment (codergs) [inline internal/commands/vaultsecrets/apps/read.go:1]: Is copyright missing?

File: internal/commands/vaultsecrets/apps/read.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+				Preamble: `Read an application:`,
+				Command: heredoc.New(ctx.IO, heredoc.WithPreserveNewlines()).Must(`
+				$ hcp vault-secrets apps read company-card
+				`),
+			},
+		},
+		Args: cmd.PositionalArguments{
+			Args: []cmd.PositionalArgument{
+				{
+					Name:          "APP_NAME",
```
Comment (codergs) [inline internal/commands/vaultsecrets/apps/read.go:55]: not sure, but should we keep this as NAME too?

File: internal/commands/vaultsecrets/apps/read_test.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+			// Run the command
+			err := readRun(opts)
+			if c.ErrMsg != "" {
+				r.Contains(err.Error(), c.ErrMsg)
+				return
+			}
+
+			r.NoError(err)
+			r.Contains(io.Output.String(), fmt.Sprintf("App Name:    %s", opts.AppName))
```
Comment (codergs) [inline internal/commands/vaultsecrets/apps/read_test.go:159]: should we keep the default format as table?

File: internal/commands/vaultsecrets/apps/read.go (hunk 2 above)
Comment (mercedesbh) [reply internal/commands/vaultsecrets/apps/read.go:55]: I think it's APP_NAME across vaultsecrets/apps. Were you referring to `NAME` somewhere else?

File: internal/commands/vaultsecrets/apps/read_test.go (hunk 3 above)
Comment (mercedesbh) [reply internal/commands/vaultsecrets/apps/read_test.go:159]: I don't have a strong opinion. This is how table format would look.
Screenshot 2024-05-14 at 10 13 51 AM

Comment (codergs) [reply internal/commands/vaultsecrets/apps/read_test.go:159]: hmmn, this is the read command. I am seeing create in the pic.

File: internal/commands/vaultsecrets/apps/read.go (hunk 2 above)
Comment (codergs) [reply internal/commands/vaultsecrets/apps/read.go:58]: I was referring to this comment - https://hashicorp.slack.com/archives/C0630MA5H8B/p1715722134440319?thread_ts=1715721973.724659&cid=C0630MA5H8B

File: internal/commands/vaultsecrets/apps/read_test.go (hunk 3 above)
Comment (mercedesbh) [reply internal/commands/vaultsecrets/apps/read_test.go:159]: Yeah, the screenshot is old, just wanted to show how the table output looks

Comment (codergs) [reply internal/commands/vaultsecrets/apps/read_test.go:159]: Please use table here as default as in RFC we are saying tabular for read commands.

File: internal/commands/vaultsecrets/apps/read.go (hunk 2 above)
Comment (mercedesbh) [reply internal/commands/vaultsecrets/apps/read.go:58]: Gotcha, will change them

**Review by codergs — APPROVED** [review]

**Pull request merged**